plaso/engine/artifact_filters.py

```diff
@@ -0,0 +1,259 @@
+  """
+
+  ARTIFACT_FILTERS = 'ARTIFACT_FILTERS'
+  COMPATIBLE_DFWINREG_KEYS = frozenset([
+      'HKEY_LOCAL_MACHINE',
+      'HKEY_LOCAL_MACHINE\\SYSTEM',
+      'HKEY_LOCAL_MACHINE\\SOFTWARE',
+      'HKEY_LOCAL_MACHINE\\SAM',
+      'HKEY_LOCAL_MACHINE\\SECURITY'])
+  STANDARD_OS_FILTERS = frozenset([
+      'windows',
+      'linux',
+      'darwin'])
+  RECURSIVE_GLOB_LIMIT = 10
+
+  def __init__(self, artifacts_registry, artifacts, knowledge_base):
+    """Initializes an artifact definitions filter helper.
+
+    Args:
+      artifacts_registry (artifacts.ArtifactDefinitionsRegistry]): artifact
+          definitions registry.
+      artifacts (list[str]): artifact names to process.
+      path (str): path to a file that contains one or more forensic artifacts.
+      knowledge_base (KnowledgeBase): Knowledge base for Log2Timeline.
+    """
+    super(ArtifactDefinitionsFilterHelper, self).__init__()
+    self._artifacts_registry = artifacts_registry
+    self._artifacts = artifacts
+    self._knowledge_base = knowledge_base
+
+  def BuildFindSpecs(self, environment_variables=None):
+    """Build find specification from a forensic artifact definitions.
+
+    Args:
+      environment_variables (Optional[list[EnvironmentVariableArtifact]]):
+          environment variables.
+    """
+    find_specs = {}
+
+    artifact_defintions = []
+    # Check if single operating system type has been provided, in that case
+    # process all artifact types for that OS.
+    if (len(self._artifacts) == 1 and
+            self._artifacts[0] in self.STANDARD_OS_FILTERS):
+      for definition in self._artifacts_registry.GetDefinitions():
+        if self._artifacts[0].lower() in (
+            os.lower() for os in definition.supported_os):
+          artifact_defintions.append(definition)
+    else:
+      for artifact_filter in self._artifacts:
+        if self._artifacts_registry.GetDefinitionByName(artifact_filter):
+          artifact_defintions.append(
+            self._artifacts_registry.GetDefinitionByName(artifact_filter))
+
+    for definition in artifact_defintions:
+      for source in definition.sources:
+        if source.type_indicator == artifact_types.TYPE_INDICATOR_FILE:
+          for path_entry in source.paths:
+            self.BuildFindSpecsFromFileArtifact(
+              path_entry, source.separator, environment_variables, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
+          keys = set(source.keys)
+          for key_entry in keys:
+            if self._CheckKeyCompatibility(key_entry):
+              self.BuildFindSpecsFromRegistryArtifact(
+                key_entry, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE):
+          # TODO: Handle Registry Values Once Supported in dfwinreg.
```

Comment:
Which dfwinreg issue is this?

Reply:
Filed https://github.com/log2timeline/dfwinreg/issues/98 and added to TODO.